lib/puppet/provider/exec/posix.rb

```diff
@@ -39,6 +42,8 @@ def checkexe(command)
   end
 
   def run(command, check = false)
+    command = splitcmd(command) unless resource[:use_shell]
```

Comment:
maybe add some description/explanation on how the command change will directly exec without shell